packages/headless/src/api/search/date/relative-date.ts

```diff
@@ -88,18 +88,31 @@ export function serializeRelativeDate(relativeDate: RelativeDate) {
   }
 }
 
+/**
+ * Deserializes relative date string value into an API format date.
+ * Throws an error if the relative date is not valid.
+ * @param date The string serialized with the format "period-amount-unit"
+ * @returns The absolute API format date.
+ */
 export function formatRelativeDateForSearchApi(date: string) {
```

Comment:
Similarly here, validation should have ideally occurred before we reach the formatting stage.